src/types/duration.js

```diff
@@ -0,0 +1,70 @@
+const ArgumentType = require('./base');
+
+class DurationArgumentType extends ArgumentType {
+	// This.duration::Milliseconds
+	constructor(client) {
+		super(client, 'duration');
+		this.timeIds = new Set(['mo', 'w', 'd', 'h', 'm', 's', 'ms']);
+		this.duration = 0;
+	}
+
+	// Confirms match before continuing.
+	// Checks if matched number is an int.
+	// Checks if matched string is in expected set.
+	validate(value) {
+		const MATCHES_ALL = value.match(/(\d+)\s*([A-Za-z]+)/g);
+
+		for(var i = 0; i < MATCHES_ALL.length; i++) {
+			var tempNum = MATCHES_ALL[i].match(/(\d+)/g);
+			var tempStr = MATCHES_ALL[i].match(/([A-Za-z]+)/g);
+			if(!tempNum || (tempNum.length !== 1)) return false;
+			if(!tempStr || (tempStr.length !== 1)) return false;
+			if(!Number.isInteger(parseInt(tempNum[0])))	return false;
+			if(!this.timeIds.has(tempStr[0])) return false;
+		}
+		return true;
+	}
+
+	// Separate each time group (Xmo, Yw, Zd, ext.)
+	// Combine to a single time value (in milliseconds)
+	// Return time value unless null
+	parse(value) {
+		const MATCHES_ALL = value.match(/(\d+)\s*([A-Za-z]+)/g);
+		var totalTime = 0;
+		MATCHES_ALL.forEach(dur => {
+			var tempNum = parseInt(dur.match(/(\d+)/g)[0]);
```

Comment:
Unexpected `var`, use `const` instead.

```suggestion
			const tempNum = parseInt(dur.match(/(\d+)/g)[0]);
```